feat: add drop_fields ingest transform (#3449)
Logflare/logflare: docs/docs.logflare.com/docs/concepts/ingestion/index.mdx

Key-value enrichment runs after field copying, so copied fields can be used as lookup sources.
:::

### Drop Fields

A source can be configured to drop specific fields from the event body before it is persisted. This is useful for keys that are needed upstream — for example, by a log shipper for routing — but should not be stored.

One field path per line. Dot syntax addresses nested paths, and `m.` is shorthand for `metadata.`. Missing paths and non-map intermediates are silently ignored.

> For example, given a Fluent Bit pipeline that uses `service` and `namespace` for routing but does not need them stored:

Reviewer: ```suggestion
For example, given a FluentBit pipeline that uses `service` and `namespace` for routing but does not need them stored:
```

Author: Held off on this one — the suggestion drops the space (`Fluent Bit` → `FluentBit`), but the canonical name is two words per fluentbit.io and the CNCF project listing. Also aligned two stale `FluentBit` references elsewhere in the codebase (marketing index alt text + pricing copy) in 0cd63356e.

```text
service
namespace
```

An incoming event:

```json
{ "service": "router", "namespace": "default", "event_message": "hello" }
```

becomes:

```json
{ "event_message": "hello" }
```

> Drop fields runs **after** field copying and key-value enrichment, so a value can be copied or used for enrichment first and then dropped from its original location:

Reviewer: ```suggestion
Drop fields runs at the end of the transformation pipeline, so a value can be copied or used for enrichment first and then dropped from its original location:
```

Author: Fixed in 3f5867714

```text
# transform_copy_fields
service:m.routing.service

# transform_drop_fields
service
```

The original `service` key is removed from the top level, but `metadata.routing.service` retains the value.

:::note ClickHouse backends

> The ClickHouse backend persists events into fixed OTEL tables via a built-in mapping that already discards unmapped top-level keys. Drop fields is still useful for ClickHouse to keep specific keys out of the catch-all `*_attributes` map columns. Be careful not to drop fields that the mapping reads (e.g. `project`, `trace_id`, `timestamp`) — those will fall back to defaults.

Reviewer: i think our clickhouse backend docs are lacking a little information on what the schema tables are for logs/metrics/traces

Author: Confirmed — `backends/clickhouse.mdx` still describes a single ingest table and pre-dates the consolidated OTEL pipeline. The `otel_logs_*` / `otel_metrics_*` / `otel_traces_*` tables, `event_type` partitioning, and catch-all `*_attributes` columns aren't documented there. Out of scope for this PR; will pick up as a follow-up.
